test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -1883,6 +1884,37 @@
            (testing "concurrent-depth metric should be 0"
              (is (= 0 (concurrent-depth))))))))))
 
+(deftest block-command-processing
+  (with-test-db
+    (svc-utils/with-pdb-with-no-gc
+      (let [dispatcher (get-service svc-utils/*server* :PuppetDBCommandDispatcher)
+            stats (-> dispatcher service-context :stats)
+            new-producer-ts (now)
+            base-cmd (get-in wire-catalogs [9 :basic])
+            enqueue-catalog (fn []
+                              (enqueue-command
+                                dispatcher
+                                (command-names :replace-catalog)
+                                9
+                                (:certname base-cmd)
+                                nil
+                                (->  base-cmd
+                                     (assoc :producer_timestamp new-producer-ts)
+                                     tqueue/coerce-to-stream)
+                                ""))]
+
+        (testing "should pause execution"
+          (pause-execution dispatcher)
+          (enqueue-catalog)
+
```

Comment:
Particularly given that CI has sometimes been *surprisingly* slow (which is how we ended up with such a high `default-ttimeout-ms`), might be worth adding another check that we're "closer" to the state we want before we block the 500ms, i.e. maybe add something like this here:
```
(is (= 1 (await-ref-state stats #(= 1 (% :received-commands))
                          default-timeout-ms :timeout))))))))
```

Reply:
I can add the check, but don't know if it make a difference because the `stats` atom is incremented in [enqueue-command ](https://github.com/puppetlabs/puppetdb/blob/b2b783d4f46daa35433e9d90645076b26faaa2b1/src/puppetlabs/puppetdb/command.clj#L1033 ) which is the first thing we do in the test `enqueue-catalog -> enqueue-command`

Reply:
re-wrote it to
``` clojure
          (is (= 1 (-> stats
                       (await-ref-state #(= 1 (% :received-commands))
                                        default-timeout-ms :timeout)
                       :received-commands))))
```